commit 08d48d49a53f0b9cede5cadbf85f345e0d648e8d (refs/changes/07/5507/1)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-02-07T12:26:30+00:00 (8 months ago)
    
    TOOLS-2176 mountain-gorilla should deal with fully specified git branch names

diff --git a/tools/jenkins-build b/tools/jenkins-build
index ed106c1..d85381a 100755
--- a/tools/jenkins-build
+++ b/tools/jenkins-build
@@ -69,6 +69,7 @@ if [[ -z "$MG_JENKINS_BUILD_SECOND_PASS" ]]; then
     # which allows, for example, a commit to a feature branch (say "foo") to
     # work when ancillary repos (like mountain-gorilla.git and usb-headnode.git)
     # don't have that branch.
+    #
     if [[ -n "$payload" ]]; then
         ref=$(echo "$payload" | json ref)
         if [[ $(echo "$ref" | cut -d/ -f2) != "heads" ]]; then
@@ -84,6 +85,22 @@ if [[ -z "$MG_JENKINS_BUILD_SECOND_PASS" ]]; then
     if [[ -z "$BRANCH" ]]; then
         BRANCH=master
     fi
+
+    #
+    # We impose a restriction on BRANCH being a "short form" branch, so
+    # require "master" rather than "refs/remotes/origin/master" by
+    # modifying the given value with /usr/bin/basename. This matches the way
+    # components compute $BRANCH from Makefile.defs. Otherwise, MG
+    # would attempt to lookup non-existent paths in Manta, e.g.
+    # /Joyent_Dev/public/builds/registrar/refs/remotes/origin/master-latest
+    # rather than /Joyent_Dev/public/builds/registrar/master-latest
+    #
+    if [[ -n "$BRANCH" ]]; then
+        BRANCH=$(/usr/bin/basename $BRANCH)
+    fi
+    if [[ -n "$TRY_BRANCH" ]]; then
+        TRY_BRANCH=$(/usr/bin/basename $TRY_BRANCH)
+    fi
     export BRANCH TRY_BRANCH
     echo "BRANCH: $BRANCH"
     echo "TRY_BRANCH: $TRY_BRANCH"
